core/src/main/java/org/elasticsearch/action/bulk/Retry.java

```diff
@@ -102,12 +125,15 @@ public BulkResponse withSyncBackoff(Client client, BulkRequest bulkRequest) thro
         private volatile BulkRequest currentBulkRequest;
         private volatile ScheduledFuture<?> scheduledRequestFuture;
 
-        AbstractRetryHandler(Class<? extends Throwable> retryOnThrowable, BackoffPolicy backoffPolicy, Client client, ActionListener<BulkResponse> listener) {
+        RetryHandler(Class<? extends Throwable> retryOnThrowable, BackoffPolicy backoffPolicy,
+                     BiConsumer<BulkRequest, ActionListener<BulkResponse>> consumer, ActionListener<BulkResponse> listener,
+                     Settings settings, BiFunction<TimeValue, Runnable, ScheduledFuture<?>> scheduleFn) {
             this.retryOnThrowable = retryOnThrowable;
             this.backoff = backoffPolicy.iterator();
-            this.client = client;
+            this.consumer = consumer;
             this.listener = listener;
-            this.logger = Loggers.getLogger(getClass(), client.settings());
+            this.logger = Loggers.getLogger(getClass(), settings);
```

Comment:
I wonder if we can get rid of settings here too. It's done to prefix the log lines with node.name. Is this still important with transport client / REST client? I think this is almost never set on a client.